src/main/java/lotto/LottoRank.java

```diff
@@ -0,0 +1,43 @@
+package lotto;
+
+
+public enum LottoRank {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    MISS(0, 0);
+
+    private final int countOfMatch;
+    private final int winningMoney;
+
+    private LottoRank(int countOfMatch, int winningMoney) {
+        this.countOfMatch = countOfMatch;
+        this.winningMoney = winningMoney;
+    }
+
+    public int getCountOfMatch() {
+        return countOfMatch;
+    }
+
+    public int getWinningMoney() {
+        return winningMoney;
+    }
+
+    public static LottoRank valueOf(int countOfMatch, boolean matchBonus) {
+        if (countOfMatch == 6) {
+            return FIRST;
+        }
+        if (countOfMatch == 5) {
+            return matchBonus ? SECOND : THIRD;
+        }
+        if (countOfMatch == 4) {
+            return FOURTH;
+        }
+        if (countOfMatch == 3) {
+            return FIFTH;
+        }
+        return MISS;
+    }
```

Comment:
```suggestion
    public static LottoRank valueOf(int countOfMatch, boolean matchBonus) {
        if(countOfMatch == 5) {
            return matchBonus ? SECOND : THIRD;
        }

        return rankBy(countOfMatch);
    }

    private static LottoRank rankBy(int countOfMatch) {
        return Arrays.stream(LottoRank.values())
                .filter(rank -> rank.countOfMatch == countOfMatch)
                .findFirst()
                .orElse(MISS);
    }
```
- if 조건을 나열하는 것보단, switch 문을 활용하는게 어떨까요? 아니라면 위와 같이 표현하는건 어떨까요?